mxfusion/modules/gp_modules/svgp_regression.py

```diff
@@ -214,60 +203,37 @@ def compute(self, F, variables):
         """
         has_mean = self.model.F.factor.has_mean
         X = variables[self.model.X]
-        N = X.shape[-2]
         Z = variables[self.model.inducing_inputs]
+        N = X.shape[-2]
         noise_var = variables[self.model.noise_var]
         mu = variables[self.graphs[1].qU_mean]
         S_W = variables[self.graphs[1].qU_cov_W]
         S_diag = variables[self.graphs[1].qU_cov_diag]
-        M = Z.shape[-2]
         kern = self.model.kernel
         kern_params = kern.fetch_parameters(variables)
 
         S = F.linalg.syrk(S_W) + make_diagonal(F, S_diag)
 
-        Kuu = kern.K(F, Z, **kern_params)
-        if self.jitter > 0.:
-            Kuu = Kuu + F.eye(M, dtype=Z.dtype) * self.jitter
+        if has_mean:
+            kern_params['mean'] = variables[self.model.mean]
 
-        L = F.linalg.potrf(Kuu)
-        Ls = F.linalg.potrf(S)
-        LinvLs = F.linalg.trsm(L, Ls)
-        Linvmu = F.linalg.trsm(L, mu)
-        LinvSLinvT = F.linalg.syrk(LinvLs)
-        wv = F.linalg.trsm(L, Linvmu, transpose=True)
+        S = F.expand_dims(S, 1)
+        mu = F.expand_dims(mu, 1)
 
-        Kxt = kern.K(F, Z, X, **kern_params)
-        mu = F.linalg.gemm2(Kxt, wv, True, False)
-        if has_mean:
-            mean = variables[self.model.mean]
-            mu = mu + mean
+        q_u = MultivariateNormalRuntime(mu[:, :, :, 0], S)
```

Comment:
Why ```mu[:, :, :, 0]```?

Reply:
So maybe this is a sign of bad design. The multivariate normal class represents the mean with shape `(n_samples, n_outputs, n_inputs)` and the covariance with shape `n_samples, n_outputs, n_inputs, n_inputs`. If we want to do linear algebra between these you end up needing the expand the dimensions of the mean so it's shape matches the covariance. Then when you go back to representing it in the `MultivariateNormal` I cut off the last dimension.

I guess some redesign is necessary here, but I'm not exactly sure where.